internal/vault/client_factory.go

```diff
@@ -249,21 +251,58 @@ func (m *cachingClientFactory) Get(ctx context.Context, client ctrlclient.Client
 		return nil, errs
 	}
 
+	ns, err := common.GetVaultNamespace(obj)
+	if err != nil {
+		return nil, err
+	}
+
+	namespacedClient := func(c Client) (Client, error) {
+		// handle the case where the "root" Client's namespace differs from that of the one specified in obj.Spec.Namespace.
+		// in which case we cache and return the namespaced Clone of the "root" Client.
+		if ns != "" && ns != c.Namespace() {
+			cacheKeyClone, err := ClientCacheKeyClone(cacheKey, ns)
+			if err != nil {
+				return nil, err
+			}
+
+			if clone, ok := m.cache.Get(cacheKeyClone); ok {
+				return clone, nil
+			}
+
+			clone, err := c.Clone(ns)
+			if err != nil {
+				return nil, err
+			}
+
+			logger.Info("Cloned Client",
```

Comment:
Does it make sense to track metrics here of the # of namespaced clients, or is it covered elsewhere?

Reply:
Do you mean including namespaces in the set of supported labels? Seems like something worth considering. I wounder if there might be any potential security issues around exposing the namespace.